src/Features/CSharp/Portable/Completion/KeywordRecommenders/AssemblyKeywordRecommender.cs

```diff
@@ -17,15 +15,22 @@ public AssemblyKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (context.IsTypeAttributeContext(cancellationToken))
+            var token = context.TargetToken;
+
+            if (token.Kind() != SyntaxKind.OpenBracketToken)
             {
-                var token = context.LeftToken;
-                var type = token.GetAncestor<MemberDeclarationSyntax>();
+                return false;
+            }
 
-                return type == null || type.IsParentKind(SyntaxKind.CompilationUnit);
+            if (token.Parent.Kind() == SyntaxKind.AttributeList)
+            {
+                var attributeList = token.Parent;
+                var previousSyntax = attributeList.Parent;
+                return previousSyntax is CompilationUnitSyntax || previousSyntax.Parent is CompilationUnitSyntax;
             }
 
-            return false;
+            var skipTrivia = token.Parent;
+            return skipTrivia.Parent == null;
```

Comment:
When do we hit this case?

Reply:
I didn't understand, What do you mean? When this condition happen?

Reply:
Yes.  That's what i'm asking.  What user code would be needed to trigger this condition being true?

Reply:
When the user code is in root file, when isn't inside namespace/class/enum/etc. When I run `TestAssemblyAttributeBeforeClass` this condition is true. But stopping to think, I think we don't need this code, I can return true direct, because only case the code will hit here is when user code is in root file.